app/controllers/claimToAdjustPoa/EnterPoAAmountController.scala

```diff
@@ -16,32 +16,99 @@
-                                           val auth: AuthenticatorPredicate,
-                                           implicit val itvcErrorHandler: ItvcErrorHandler,
-                                           implicit val itvcErrorHandlerAgent: AgentItvcErrorHandler)
-                                          (implicit val appConfig: FrontendAppConfig,
-                                           implicit override val mcc: MessagesControllerComponents,
-                                           val ec: ExecutionContext)
-  extends ClientConfirmedController {
+                                         val auth: AuthenticatorPredicate,
+                                         val sessionService: PaymentOnAccountSessionService,
+                                         view: EnterPoAAmountView,
+                                         claimToAdjustService: ClaimToAdjustService,
+                                         implicit val itvcErrorHandler: ItvcErrorHandler,
+                                         implicit val itvcErrorHandlerAgent: AgentItvcErrorHandler)
+                                        (implicit val appConfig: FrontendAppConfig,
+                                         implicit override val mcc: MessagesControllerComponents,
+                                         val ec: ExecutionContext)
+  extends ClientConfirmedController with FeatureSwitching with ClaimToAdjustUtils {
 
   def show(isAgent: Boolean): Action[AnyContent] =
     auth.authenticatedAction(isAgent) {
       implicit user =>
-        Future successful Ok(
-          s"to be implemented: /report-quarterly/income-and-expenses/view/${if (isAgent) "agents" else ""}adjust-poa/enter-poa-amount")
+        ifAdjustPoaIsEnabled(isAgent) {
+          withvalidSession(isAgent) { _ =>
+            claimToAdjustService.getEnterPoAAmountViewModel(Nino(user.nino)).map {
+              case Right(viewModel) =>
+                Ok(view(EnterPoaAmountForm.form, viewModel, isAgent, controllers.claimToAdjustPoa.routes.EnterPoAAmountController.submit(isAgent)))
+              case Left(ex) =>
+                Logger("application").error(s"Error while retrieving charge history details : ${ex.getMessage} - ${ex.getCause}")
+                showInternalServerError(isAgent)
+            }
+          }
+        }.recover {
+          case ex: Exception =>
+            Logger("application").error(s"Unexpected error: ${ex.getMessage} - ${ex.getCause}")
+            showInternalServerError(isAgent)
+        }
     }
 
+  def submit(isAgent: Boolean): Action[AnyContent] = auth.authenticatedAction(isAgent) {
+    implicit request =>
+      ifAdjustPoaIsEnabled(isAgent) {
+        claimToAdjustService.getEnterPoAAmountViewModel(Nino(request.nino)).flatMap {
+          case Right(viewModel) =>
+            EnterPoaAmountForm.checkValueConstraints(EnterPoaAmountForm.form.bindFromRequest(), viewModel.totalAmountOne, viewModel.relevantAmountOne).fold(
```

Comment:
might worth to move whole EnterPoaAmountForm.checkValueConstraints .. section under private method, tidy refactoring to improve readability